packages/dds/tree/docs/data-model/README.md

```diff
@@ -0,0 +1,166 @@
+
+<figure align="center">
+  <img src="./assets/root.drawio.svg" alt=""/>
+  <figcaption>Figure: Implicit root node</figcaption>
+</figure>
+
+### Value
+
+Each tree node has an optional ***value***.
+Values are used to store scalar data, such as numbers and booleans.
+
+<figure align="center">
+  <img src="./assets/scalar-nodes.drawio.svg" alt=""/>
+  <figcaption>Figure: Nodes with values</figcaption>
+</figure>
+
+From the perspective of the SharedTree data model, values are opaque.
+The only tree operation that affects a node's value is 'setValue', which overwrites a node's value with a new opaque value.
+
+From the perspective of the broader Fluid system, node values are [*serializable*](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/datastore-definitions/src/serializable.ts)
+and are traversed by Fluid services such as garbage collection.
+
+### Field
+
+Each tree node has a set of zero or more fields.
+Fields are used to model map-like composite type, where each  ***field*** represents one relationship between a parent node and an ordered ***sequence*** of one or more children.
+
+<figure align="center">
+  <img src="./assets/children.drawio.svg" alt=""/>
+  <figcaption>Figure: A parent node with two fields</figcaption>
+</figure>
+
+The fields of a node are distinguished by a ***key***.
+Fields of the same node are distinguished by a field key.
+From the data model's perspective, field keys are opaque.
+
+### Sequences
+
+In the SharedTree data model abstraction, there is no distinction between a field that always contain a single child and a field that can contain multiple children.
+Instead, SharedTree treats all fields as ordered collections and uses schema to restrict which fields are optional, must contain a single value, or may contain multiple values.
+
+When mapping the SharedTree data model to conventional programming languages, it is sometimes helpful to think of the sequence as being distinct from the field, as if each field points to an ***implicit sequence*** object.
+
+However, it is important to remember that sequences are not tree nodes.
+This means that sequences are not directly addressable.
+The combination of the field + sequences is implicitly created when the first item is inserted, implicitly deleted when the last item is removed, and may only be referenced indirectly via the combination of parent node + field key.
+
+### Special Fields
+
+This section covers fields that receive special treatment in the SharedTree data model.
+These fields are special because:
+
+- They have well-known keys
+- They are universally available on all nodes (regardless of schema.)
+- They can not be targeted by normal tree operations.
+
+#### Type
+
+The SharedTree data model supports nominal typing via a special *type* field.
+The value of the *type* field is the unique identifier of the corresponding schema type.
+
+In the data model abstraction, the value of the type field is opaque.
```

Comment:
Okay, I think at this point it is worth defining what "opaque with respect to the data model" means early on in this document and then all these "opaque" comments will fall out :)